README.md
<img src="src/icon.svg" alt="icon" width="100" height="100">

# Back to Top plugin for Craft CMS 3 

Add Back to top to your website

## Installation

#### Requirements

This plugin requires Craft CMS 3.0.0, or later.

#### Plugin Store

Log into your control panel and click on 'Plugin Store'. Search for 'Back to Top'.
1. Open your terminal and go to your Craft project:

```bash
cd /path/to/project
```

2. Then tell Composer to load the plugin:

```bash
composer require ds/craft-backtotop
```

3. In the Control Panel, go to Settings → Plugins and click the “Install” button for Back to top.



CHANGELOG.md
# Back to Top Changelog

All notable changes to this project will be documented in this file.

## 1.1- 2022-02-10
### Updated Few Changes
- Updated Few Changes

## 1.0.0 - 2022-01-17
### Added
- Initial release


src/dsbacktotop.php
<?php
/**
 * DS Back To Top plugin for Craft CMS 3.x
 *
 * DS BacktoTop
 *
 * @link      https://www.dotsquares.com/
 * @copyright Copyright (c) 2021 Dotsquares
 */

namespace ds\dsbacktotop;

use Craft;
use craft\base\Plugin;
use craft\services\Plugins;
use ds\dsbacktotop\models\Settings;
use ds\dsbacktotop\services\dsbacktotopService;
use craft\web\twig\variables\CraftVariable;
use ds\dsbacktotop\variables\dsbacktotopVariable;

use yii\base\Event;



class dsbacktotop extends Plugin

{
    public $hasCpSettings = true;
	public static $plugin;
    
    // Public Methods
    // =========================================================================

    public function init()
    {
        parent::init();
         self::$plugin = $this;
         
        // Register our service

        $this->setComponents([
            //classname
            'dsbacktotop' =>  services\dsbacktotopService::class,
            
        ]);

        
        // Register our variable
        Event::on(
            CraftVariable::class,
            CraftVariable::EVENT_INIT,
            function (Event $event) {
                /** @var CraftVariable $variable */
                $variable = $event->sender;
                //handle name                             //classname
                $variable->set('dsbacktotop', variables\dsbacktotopVariable::class);
            }
        );


       
    }

     // Protected Methods
    // =========================================================================
        protected function createSettingsModel()
        {
            return new Settings();
        }
    
    
            // Get the settings that are being defined by template
    
        protected function settingsHtml()
        {
            return \Craft::$app->getView()->renderTemplate(
                // Plugin handle name/settings
                'dsbacktotop/settings',
                [ 'settings' => $this->getSettings() ]
            );

        } 
    
}
?>

src/models/Settings.php
<?php
/**
 * DS Back To Top  plugin for Craft CMS 3.x
 *
 * DS BacktoTop
 *
 * @link      https://www.dotsquares.com/
 * @copyright Copyright (c) 2021 Dotsquares
 */

namespace ds\dsbacktotop\models;
use ds\dsbacktotop\dsbacktotop;
use Craft;
use craft\base\Model;

/**
 * DS BacktoTop Model
 *
 * Models are containers for data. Just about every time information is passed
 * between services, controllers, and templates in Craft, it’s passed via a model.
 
 */


class Settings extends Model
{
   // Public Properties
    // =========================================================================

   
    
    public $onsite;
    public $bgcolor;
    public $bgwidth;
    public $bgheight;
    public $fontsize;
    public $hovercolor;
    public $scrollduration;
    public $position;
    public $select;
    
   
    // Returns the validation rules for attributes.

    

  
    public function getonsite()
    {     
        return Craft::parseEnv(trim(dsbacktotop::getInstance()->getSettings()->onsite));
    }

    public function getfavcolor()
    {     
        return Craft::parseEnv(trim(dsbacktotop::getInstance()->getSettings()->bgcolor));
    }

    public function getbgwidth()
    {     
        return Craft::parseEnv(trim(dsbacktotop::getInstance()->getSettings()->bgwidth));
    }
    public function getbgheight()
    {     
        return Craft::parseEnv(trim(dsbacktotop::getInstance()->getSettings()->bgheight));
    }
    public function getfontsize()
    {     
        return Craft::parseEnv(trim(dsbacktotop::getInstance()->getSettings()->fontsize));
    }

    public function gethovercolor()
    {     
        return Craft::parseEnv(trim(dsbacktotop::getInstance()->getSettings()->hovercolor));
    }

    public function getscrollduration()
    {     
        return Craft::parseEnv(trim(dsbacktotop::getInstance()->getSettings()->scrollduration));
    }

    public function getposition()
    {     
        return Craft::parseEnv(trim(dsbacktotop::getInstance()->getSettings()->position));
    }

    public function getselect()
    {     
        return Craft::parseEnv(trim(dsbacktotop::getInstance()->getSettings()->select));
    }
    
   
    
    
}

src/services/dsbacktotopService.php
<?php
/**
 * DS Back To Top plugin for Craft CMS 3.x
 *
 * DS BacktoTop
 * 
 * @link      https://www.dotsquares.com/
 * @copyright Copyright (c) 2021 Dotsquares
 */


namespace ds\dsbacktotop\services;
use Craft;
use craft\base\Component;
use ds\dsbacktotop\dsbacktotop;
use ds\dsbacktotop\resources\dsbacktotopAsset;

/**
 * DS BacktoTop Service
 *
 * All of your plugin’s business logic should go in services, including saving data,
 * retrieving data, etc. They provide APIs that your controllers, template variables,
 * and other plugins can interact with.
 *
 * https://craftcms.com/docs/plugins/services
 *
 * @author    Dotsquares
 */


class dsbacktotopService extends  Component

{
  
    // Public Methods
    // =========================================================================

    public function dsbacktotop(){
       
      

      // get setting main file
      $settings = dsbacktotop::getInstance()->getSettings();

      $enbledsites  = $settings->getonsite();
      $backgroundcolor  = $settings->getfavcolor();
      $bgwidth  = $settings->getbgwidth();
      $bgheight  = $settings->getbgheight();
      $fontsizes  = $settings->getfontsize();
      $hovercolor  = $settings->gethovercolor();
      $scrollduration  = $settings->getscrollduration();
      $position  = $settings->getposition();
      $select  = $settings->getselect();

      echo "<link rel='stylesheet' href='https://pro.fontawesome.com/releases/v5.10.0/css/all.css'/>";
      
?>

<?php 
if($enbledsites=='1'){ ?>
<a href="#" class="scrollup waves-effect waves-dark" style="<?php if($backgroundcolor){ ?>background:<?php echo $backgroundcolor; }?>;
<?php if($bgwidth){ ?>width:<?php echo $bgwidth.'px'; } ?>;<?php if($bgheight){ ?>height:<?php echo $bgheight.'px'; } ?>;
<?php if($position){ ?><?php echo $position ?>:20px;<?php }?>">


<?php if($select=='labeltype1'){?>
  <i <?php if($fontsizes){ ?> style="font-size:<?php echo $fontsizes.'px'; ?>; margin-top:20px;" <?php } ?> class="fa fa-arrow-up fa-lg">
 <?php } elseif($select=='labeltype2'){?>
  <i <?php if($fontsizes){ ?> style="font-size:<?php echo $fontsizes.'px'; ?>" <?php } ?> class="fa fa-level-up fa-2x">
 <?php } elseif($select=='labeltype3'){?>
  <i <?php if($fontsizes){ ?> style="font-size:<?php echo $fontsizes.'px'; ?>" <?php } ?> class="fa fa-angle-double-up fa-2x">
 <?php } elseif($select=='labeltype4'){?>
  <i <?php if($fontsizes){ ?> style="font-size:<?php echo $fontsizes.'px'; ?>" <?php } ?>class="fa fa-reply fa-rotate-90 fa-2x">
 <?php }else{?>
  <i <?php if($fontsizes){ ?> style="font-size:<?php echo $fontsizes.'px'; ?>" <?php } ?>class="fa fa-angle-up ds">
  <?php }?>
 </i>
</a>

 <?php } ?>

<style>

.scrollup{
    position:fixed;
    width:32px;
    height:auto;
    bottom:6px;
    right:20px;
    display:none;
    background-color:#ff9800;
    z-index: 99999;
    padding-bottom: 6px;
}

a.scrollup {
	outline:0;
	text-align: center;
}

a.scrollup:hover,a.scrollup:active,a.scrollup:focus {
	opacity:0.5;
	text-decoration:none;
  background:<?php if($hovercolor){  echo $hovercolor.'!important'; } ?>
  
}
a.scrollup i {
	margin-top: 10px;
	color: #fff;
}
a.scrollup i:hover {
	text-decoration:none;
}
</style>
<script src="https://code.jquery.com/jquery-2.1.1.min.js" type="text/javascript"></script> 
<script>
$(window).scroll(function(){
  
			if ($(this).scrollTop() > 100) {
				$('.scrollup').fadeIn();
				} else {
				$('.scrollup').fadeOut();
			}
		});
		$('.scrollup').click(function(){
			$("html, body").animate({ scrollTop: 0 }, <?php if($scrollduration){echo $scrollduration;}else{ echo '1500';}?>);
				return false;
		});

  </script>
  

<?php     
}

}

src/services/old--dsbacktotopService.php
<?php
/**
 * DS Back To Top plugin for Craft CMS 3.x
 *
 * DS BacktoTop
 * 
 * @link      https://www.dotsquares.com/
 * @copyright Copyright (c) 2021 Dotsquares
 */


namespace ds\dsbacktotop\services;
use Craft;
use craft\base\Component;
use ds\dsbacktotop\dsbacktotop;
use ds\dsbacktotop\resources\dsbacktotopAsset;

/**
 * DS BacktoTop Service
 *
 * All of your plugin’s business logic should go in services, including saving data,
 * retrieving data, etc. They provide APIs that your controllers, template variables,
 * and other plugins can interact with.
 *
 * https://craftcms.com/docs/plugins/services
 *
 * @author    Dotsquares
 */


class dsbacktotopService extends  Component

{
  
    // Public Methods
    // =========================================================================

    public function dsbacktotop(){
       
      

      // get setting main file
      $settings = dsbacktotop::getInstance()->getSettings();

      $enbledsites  = $settings->getonsite();
      $backgroundcolor  = $settings->getfavcolor();
      $bgwidth  = $settings->getbgwidth();
      $bgheight  = $settings->getbgheight();
      $fontsizes  = $settings->getfontsize();
      $hovercolor  = $settings->gethovercolor();
      
      $scrollduration  = $settings->getscrollduration();
      $position  = $settings->getposition();
      $options_one  = $settings->getlabeltype1();
      $options_second  = $settings->getlabeltype2();
      $options_third  = $settings->getlabeltype3();
      $options_fourth  = $settings->getlabeltype4();
      $select  = $settings->getselect();
     //echo '@@@';print_r($select);

      echo "<link rel='stylesheet' href='https://pro.fontawesome.com/releases/v5.10.0/css/all.css'/>";
      
?>

<?php 
if($enbledsites=='1'){ ?>
<a href="#" class="scrollup waves-effect waves-dark" style="<?php if($backgroundcolor){ ?>background:<?php echo $backgroundcolor; }?>;
<?php if($bgwidth){ ?>width:<?php echo $bgwidth.'px'; } ?>;<?php if($bgheight){ ?>height:<?php echo $bgheight.'px'; } ?>;
<?php if($position){ ?><?php echo $position ?>:20px;<?php }?>">


<?php if($select=='labeltype1'){?>
  <i <?php if($fontsizes){ ?> style="font-size:<?php echo $fontsizes.'px'; ?>; margin-top:20px;" <?php } ?> class="fa fa-arrow-up fa-lg">
 <?php } elseif($select=='labeltype2'){?>
  <i <?php if($fontsizes){ ?> style="font-size:<?php echo $fontsizes.'px'; ?>" <?php } ?> class="fa fa-level-up fa-2x">
 <?php } elseif($select=='labeltype3'){?>
  <i <?php if($fontsizes){ ?> style="font-size:<?php echo $fontsizes.'px'; ?>" <?php } ?> class="fa fa-angle-double-up fa-2x">
 <?php } elseif($select=='labeltype4'){?>
  <i <?php if($fontsizes){ ?> style="font-size:<?php echo $fontsizes.'px'; ?>" <?php } ?>class="fa fa-reply fa-rotate-90 fa-2x">
 <?php }else{?>
  <i <?php if($fontsizes){ ?> style="font-size:<?php echo $fontsizes.'px'; ?>" <?php } ?>class="fa fa-angle-up ds">
  <?php }?>
 </i>
</a>

 <?php } ?>

<style>

.scrollup{
    position:fixed;
    width:32px;
    height:32px;
    bottom:0px;
    right:20px;
    display:none;
    background-color:#ff9800;
    z-index: 99999;
}

a.scrollup {
	outline:0;
	text-align: center;
}

a.scrollup:hover,a.scrollup:active,a.scrollup:focus {
	opacity:0.5;
	text-decoration:none;
  background:<?php if($hovercolor){  echo $hovercolor.'!important'; } ?>
  
}
a.scrollup i {
	margin-top: 10px;
	color: #fff;
}
a.scrollup i:hover {
	text-decoration:none;
}
</style>
<script src="https://code.jquery.com/jquery-2.1.1.min.js" type="text/javascript"></script> 
<script>
$(window).scroll(function(){
  
			if ($(this).scrollTop() > 100) {
				$('.scrollup').fadeIn();
				} else {
				$('.scrollup').fadeOut();
			}
		});
		$('.scrollup').click(function(){
			$("html, body").animate({ scrollTop: 0 }, <?php echo $scrollduration;?>);
				return false;
		});

  </script>
  

<?php     
}

}